components/chassis.py

```diff
@@ -114,6 +114,10 @@ def get_pose(self) -> Pose2d:
         """Get the current position of the robot on the field."""
         return self.odometry.getPose()
 
+    def get_heading(self) -> float:
+        """Get the current heading of the robot."""
+        return self.get_pose().rotation().radians()
```

Comment:
I'm inclined to have this return the Rotation2d instead. Its addition/subtraction implementation bounds the radians value to [-pi,pi], which looks like it might be useful below.

Reply:
The reason for this is the amount of boilerplate required to get the heading when you want it. Which tends to be what people want - the upstream API is one that is technically the correct way to do it, and wholly unhelpful for the vast majority of uses!

Reply:
Also, I can change to doing all the differences with `Pose2d` or `Rotation2d` objects, but I don't think it will reduce the number of calls to a constraining function - the geometry library must be making that call every time a difference is calculated any way.
tl;dr - I'll change this if the consensus is that it's a better way to do it.

Reply:
Might not be worth it unless we use it consistently. Maybe let's see how we go. If we think this has become too messy we can see how swapping everything over goes.